tests/test_Vitocal252A.py

```diff
@@ -0,0 +1,253 @@
+        self.assertEqual(
+            self.device.circuits[0].getHeatingCurveShift(), 0)
+
+    def test_getReturnTemperature(self):
+        self.assertEqual(self.device.getReturnTemperature(), 39.3)
+
+    def test_getSupplyTemperaturePrimaryCircuit(self):
+        self.assertEqual(
+            self.device.getSupplyTemperaturePrimaryCircuit(), 3.0)
+
+    def test_getPrograms(self):
+        expected_programs = ['comfortCooling', 'comfortCoolingEnergySaving', 'comfortEnergySaving', 'comfortHeating', 'fixed', 'forcedLastFromSchedule', 'frostprotection', 'normalCooling', 'normalCoolingEnergySaving', 'normalEnergySaving', 'normalHeating', 'reducedCooling', 'reducedCoolingEnergySaving', 'reducedEnergySaving', 'reducedHeating', 'standby']
+        self.assertListEqual(expected_programs, self.device.circuits[0].getPrograms())
+
+    def test_getModes(self):
+        expected_modes = ['heating', 'standby']
+        self.assertListEqual(expected_modes, self.device.circuits[0].getModes())
+
+    # Total power consumption:
+    def test_getPowerConsumptionUnit(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionUnit(), "kilowattHour")
+
+    def test_getPowerConsumptionToday(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionToday(), 13.0)
+
+    def test_getPowerConsumptionThisMonth(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionThisMonth(), 176.10000000000002)
+
+    def test_getPowerConsumptionThisYear(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionThisYear(), 478.6)
+
+    # Power consumption for Domestic Hot Water:
+    def test_getPowerConsumptionDomesticHotWaterUnit(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionDomesticHotWaterUnit(), "kilowattHour")
+
+    def test_getPowerConsumptionDomesticHotWaterToday(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionDomesticHotWaterToday(), 2.3)
+
+    def test_getPowerConsumptionDomesticHotWaterThisMonth(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionDomesticHotWaterThisMonth(), 45.2)
+
+    def test_getPowerConsumptionDomesticHotWaterYear(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionDomesticHotWaterYear(), 215.4)
+
+    # Power consumption for Heating:
+    def test_getPowerConsumptionHeatingUnit(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionHeatingUnit(), "kilowattHour")
+
+    def test_getPowerConsumptionHeatingToday(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionHeatingToday(), 10.7)
+
+    def test_getPowerConsumptionHeatingThisMonth(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionHeatingThisMonth(), 130.9)
+
+    def test_getPowerConsumptionHeatingYear(self):
+        self.assertEqual(
+            self.device.getPowerConsumptionHeatingYear(), 263.2)
```

Comment:
```suggestion
    def test_getPowerConsumptionHeatingThisYear(self):
        self.assertEqual(
            self.device.getPowerConsumptionHeatingThisYear(), 263.2)
```

Reply:
As, this is a total mess, there are already method that are names this way, but also the other way 🤯